src/com/maddyhome/idea/vim/group/visual/VisualGroup.kt

```diff
@@ -0,0 +1,252 @@
+/*
+ * IdeaVim - Vim emulator for IDEs based on the IntelliJ platform
+ * Copyright (C) 2003-2019 The IdeaVim authors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.maddyhome.idea.vim.group.visual
+
+import com.intellij.openapi.editor.*
+import com.intellij.openapi.editor.colors.EditorColors
+import com.maddyhome.idea.vim.VimPlugin
+import com.maddyhome.idea.vim.command.CommandState
+import com.maddyhome.idea.vim.group.MotionGroup
+import com.maddyhome.idea.vim.helper.EditorHelper
+import com.maddyhome.idea.vim.helper.sort
+import com.maddyhome.idea.vim.helper.vimLastColumn
+import com.maddyhome.idea.vim.helper.vimSelectionStart
+
+/**
+ * @author Alex Plate
+ */
+
+/**
+ * Set selection for caret
+ * This method doesn't change CommandState and operates only with caret and it's properties
+ * if [moveCaretToSelectionEnd] is true, caret movement to [end] will be performed
+ */
+fun Caret.vimSetSelection(start: Int, end: Int = start, moveCaretToSelectionEnd: Boolean = false) {
```

Comment:
I think for methods it's better to follow the word order for English sentences: "set vim selection" rather than to use "vim" as a prefix. See also the methods below.